pandas/core/strings.py

```diff
@@ -464,11 +464,9 @@ def rep(x, r):
         return result
 
 
-def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=False):
+def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=None):
```

Comment:
ok, this should have been changed a long time ago. no reason to keep a dead API around.

and change to ``FutureWarning``. can remove in next major version.